src/aof.c

```diff
@@ -1556,7 +1556,7 @@ int rewriteAppendOnlyFile(char *filename) {
 
     if (server.aof_use_rdb_preamble) {
         int error;
-        if (rdbSaveRio(SLAVE_REQ_NONE,&aof,&error,RDBFLAGS_AOF_PREAMBLE,NULL) == C_ERR) {
+        if (rdbSaveRio(SLAVE_REQ_RDB_ALL,&aof,&error,RDBFLAGS_AOF_PREAMBLE,NULL) == C_ERR) {
```

Comment:
i'm not sure i'm a fan of that ALL flag, and not sure why you use the DEFAULT in some places and ALL in others.
i think there's a chance we'll some day add requirements that are not about rdb filtering.

maybe the right way to think of it is that NONE means there are no special requirements.
then each flag in that register can add a requirement, some requirements could mean to ask for things that are included by default to be filtered, and other can maybe ask to include things that are not included by default.

i.e. the requirement flag is not a direct mapping of the `rdb-filter-only` replconf.

Reply:
I think I agree with what you wrote and you might not have understood what I meant with these defines:
```c
/* Slave requirements */
#define SLAVE_REQ_RDB_DATA (1 << 0)
#define SLAVE_REQ_RDB_FUNCTIONS (1 << 1)
#define SLAVE_REQ_RDB_ALL (SLAVE_REQ_RDB_DATA | SLAVE_REQ_RDB_FUNCTIONS)
#define SLAVE_REQ_DEFAULT SLAVE_REQ_RDB_ALL
```
`SLAVE_REQ_RDB_DATA` - Slave is requesting the dataset in the RDB.
`SLAVE_REQ_RDB_FUNCTIONS` - Slave is requesting the functions in the RDB.
`SLAVE_REQ_RDB_ALL` - The subset of requirements currently used to indicate we want all the RDB (functions, data, etc) this is just a bit-wise OR of all the `SLAVE_REQ_RDB_*` fields. Any where in the code where we want a complete RDB we'll use this (for example `SAVE` command).
`SLAVE_REQ_DEFAULT` - Is the default requirements for replication assuming we didn't receive any `REPLCONF` that changes them. For now this is the same as `SLAVE_REQ_RDB_ALL` but in the future it might also include other bits which aren't related to RDB saving.